sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -1584,6 +1584,234 @@ class CollationSQLExpressionsSuite
+        s"""
+          |select trunc(collate('2016-12-31 23:59:59', '${collationName}'), 'MM')
+          |""".stripMargin
+      // Result
+      val testQuery = sql(query)
+      val dataType = DateType
+      val expectedResult = "2016-12-01"
+      assertResult(dataType)(testQuery.schema.fields.head.dataType)
+      assertResult(expectedResult)(testQuery.collect().head.getDate(0).toString)
+    })
+  }
+
+  test("TruncTimestamp expression with collation") {
+    // Supported collations
+    Seq("UTF8_BINARY", "UTF8_BINARY_LCASE", "UNICODE", "UNICODE_CI").foreach(collationName => {
+      val query =
+        s"""
+          |select date_trunc(collate('HOUR', '${collationName}'),
+          |collate('2015-03-05T09:32:05.359', '${collationName}'))
+          |""".stripMargin
+      // Result
+      val testQuery = sql(query)
+      val dataType = TimestampType
+      val expectedResult = "2015-03-05 09:00:00.0"
+      assertResult(dataType)(testQuery.schema.fields.head.dataType)
+      assertResult(expectedResult)(testQuery.collect().head.getTimestamp(0).toString)
+    })
+  }
+
+  test("MakeTimestamp expression with collation") {
+    // Supported collations
+    Seq("UTF8_BINARY", "UTF8_BINARY_LCASE", "UNICODE", "UNICODE_CI").foreach(collationName => {
+      val query =
+        s"""
+          |select make_timestamp(2014, 12, 28, 6, 30, 45.887, collate('CET', '${collationName}'))
+          |""".stripMargin
+      // Result
+      val testQuery = sql(query)
+      val dataType = TimestampType
+      val expectedResult = "2014-12-27 21:30:45.887"
+      assertResult(dataType)(testQuery.schema.fields.head.dataType)
+      assertResult(expectedResult)(testQuery.collect().head.getTimestamp(0).toString)
+    })
+  }
+
+  test("DatePart expression with collation") {
+    // Supported collations
+    Seq("UTF8_BINARY", "UTF8_BINARY_LCASE", "UNICODE", "UNICODE_CI").foreach(collationName => {
+      val query =
+        s"""
+          |select date_part(collate('Week', '${collationName}'),
+          |collate('2019-08-12 01:00:00.123456', '${collationName}'))
+          |""".stripMargin
+      // Result
+      val testQuery = sql(query)
+      val dataType = IntegerType
+      val expectedResult = 33
+      assertResult(dataType)(testQuery.schema.fields.head.dataType)
+      assertResult(expectedResult)(testQuery.collect().head.getInt(0))
+    })
+  }
+
+  test("ConvertTimezone expression with collation") {
+    // Supported collations
+    Seq("UTF8_BINARY", "UTF8_BINARY_LCASE", "UNICODE", "UNICODE_CI").foreach(collationName => {
+      val query =
+        s"""
+          |select convert_timezone(collate('America/Los_Angeles', '${collationName}'),
+          |collate('UTC', '${collationName}'), collate('2021-12-06 00:00:00', '${collationName}'))
+          |""".stripMargin
+      // Result
+      val testQuery = sql(query)
+      val dataType = TimestampNTZType
+      assertResult(dataType)(testQuery.schema.fields.head.dataType)
```

Comment:
for all tests, we need to check both query result value, as well as result data type (like other tests from this suite)